independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Beans.java

```diff
@@ -119,6 +124,27 @@ static BeanInfo createClassBean(ClassInfo beanClass, BeanDeployment beanDeployme
         return bean;
     }
 
+    private static ScopeInfo inheritScope(ClassInfo beanClass, BeanDeployment beanDeployment) {
+        DotName superClassName = beanClass.superName();
+        while (!superClassName.equals(DotNames.OBJECT)) {
+            ClassInfo classFromIndex = beanDeployment.getIndex().getClassByName(superClassName);
+            if (classFromIndex == null) {
```

Comment:
@mkouba so the question here is what should we do if the class wasn't indexed. We can:
* blow up
* log warning and continue (which means it will be `@Dependent`)
* just silently continue

Reply:
On second thought, blowing up might not be the best because even if the class was indexed, there might be no scope to inherit and it would naturally end up as `@Dependent`.

Reply:
With the above in mind, I added logging and let it continue with dependent.